src/components/checkbox/index.tsx

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable jsx-a11y/label-has-for */
+import React, {
+  memo,
+  FC,
+  useState,
+  PropsWithChildren,
+  useEffect,
+  MouseEventHandler
+} from 'react';
+import { compose } from 'redux';
+import cuid from 'cuid';
+import { LabelPosition, Variant } from './enums';
+
+import SC from './styled';
+
+interface Props {
+  variant?: Variant;
+  checked?: boolean;
+  labelPosition?: LabelPosition;
+  onClick: (checked: boolean) => void;
+}
+
+const Checkbox: FC<Props> = ({
```

Comment:
Det kan være greit å wrappe Props med PropsWithChildren for å unngå klaging fra TS på children prop